Data/Ring/Ordered.hs

```diff
@@ -0,0 +1,377 @@
+    (-1) -> Negative
+    0 -> Zero
+    _ -> Positive
+  signum' = Num.signum
+
+-- | Where @a@ is a \'true\' or \'mathematical\' ordered ring, so is this.
+--
+-- @since 0.7
+deriving instance (OrderedRing a) => OrderedRing (Down a)
+
+-- | Where @a@ is a \'true\' or \'mathematical\' ordered ring, so is this.
+--
+-- @since 0.7
+deriving instance (OrderedRing a) => OrderedRing (Identity a)
+
+-- | @since 0.7
+instance (HasResolution a) => OrderedRing (Fixed a) where
+  abs = Num.abs
+  signum x = case Num.signum x of 
+    (-1) -> Negative
+    0 -> Zero
+    _ -> Positive
+  signum' = Num.signum
+
+-- | @since 0.7
+instance OrderedRing Int8 where
+  abs = Num.abs
+  signum x = case Num.signum x of 
+    (-1) -> Negative
+    0 -> Zero
+    _ -> Positive
+  signum' = Num.signum
+
+-- | @since 0.7
+instance OrderedRing Int16 where
+  abs = Num.abs
+  signum x = case Num.signum x of 
+    (-1) -> Negative
+    0 -> Zero
+    _ -> Positive
+  signum' = Num.signum
+
+-- | @since 0.7
+instance OrderedRing Int32 where
+  abs = Num.abs
+  signum x = case Num.signum x of 
+    (-1) -> Negative
+    0 -> Zero
+    _ -> Positive
+  signum' = Num.signum
+
+-- | @since 0.7
+instance OrderedRing Int64 where
+  abs = Num.abs
+  signum x = case Num.signum x of 
+    (-1) -> Negative
+    0 -> Zero
+    _ -> Positive
+  signum' = Num.signum
+
+-- | @since 0.7
+instance OrderedRing Int where
+  abs = Num.abs
+  signum x = case Num.signum x of 
+    (-1) -> Negative
+    0 -> Zero
+    _ -> Positive
+  signum' = Num.signum
+
+-- | This instance is a \'true\' or \'mathematical\' ordered ring, as 'Integer'
+-- is an infinite type.
+--
+-- @since 0.7
+instance OrderedRing Integer where
+  abs = Num.abs
+  signum x = case Num.signum x of 
+    (-1) -> Negative
+    0 -> Zero
+    _ -> Positive
+  signum' = Num.signum
+
+-- | @since 0.7
+instance OrderedRing (Modular Word8) where
+  abs x = x
+  signum x 
+    | x == zero = Zero
+    | otherwise = Positive
+  signum' (Modular x) = Modular . Num.signum $ x
+
+-- | @since 0.7
+instance OrderedRing (Modular Word16) where
+  abs x = x
+  signum x 
+    | x == zero = Zero
+    | otherwise = Positive
+  signum' (Modular x) = Modular . Num.signum $ x
+
+-- | @since 0.7
+instance OrderedRing (Modular Word32) where
+  abs x = x
+  signum x 
+    | x == zero = Zero
+    | otherwise = Positive
+  signum' (Modular x) = Modular . Num.signum $ x
+
+-- | @since 0.7
+instance OrderedRing (Modular Word64) where
+  abs x = x
+  signum x 
+    | x == zero = Zero
+    | otherwise = Positive
+  signum' (Modular x) = Modular . Num.signum $ x
+
+-- | @since 0.7
+instance OrderedRing (Modular Word) where
+  abs x = x
+  signum x 
+    | x == zero = Zero
+    | otherwise = Positive
+  signum' (Modular x) = Modular . Num.signum $ x
+
+-- | Where @a@ and @b@ are both \'true\' or \'mathematical\' ordered rings, so is 
+-- this.
+--
+-- @since 0.7
+instance (OrderedRing a, OrderedRing b) => 
+  OrderedRing (a, b) where
```

Comment:
This does not look right. Let `a ~ b ~ Double`, so `(a, b)` is a plane. The instance defines that the first and the third quadrants are "positive", and the second and the fourth are "negative", while both axis are "zeros". This is not a total order.

Reply:
`Double`s aren't `OrderedRings`, because their `Ord` instance is a lie and a delusion. However, in general this does hold, so I'll remove those instances.